lib/cucumber/formatter/junit.rb

```diff
@@ -25,155 +21,134 @@ def initialize(runtime, io, options)
         @options = options
       end
 
-      def before_feature(feature)
-        @current_feature = feature
-        @failures = @errors = @tests = @skipped = 0
-        @builder = Builder::XmlMarkup.new( :indent => 2 )
-        @time = 0
+      def before_test_case(test_case)
+        if not same_feature_as_previous_test_case?(test_case.feature)
```

Comment:
In order to hear about code conventions, you can use [Rubocop](https://github.com/bbatsov/rubocop) on the files you've changed, and get a detailed report on any code-style deviations.